bal_addresses/queries.py

```diff
@@ -0,0 +1,155 @@
+    blocks = defaultdict(lambda: None, blocks)
+
+    for chain in AddrBook.chain_ids_by_name.keys():
+        ## Mainnet often has a different URL string format
+        if chain == "mainnet":
+            balancer[chain] = "https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-v2"
+            gauges[chain] = "https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-gauges"
+            aura[chain] = "https://graph.aura.finance/subgraphs/name/aura/aura-mainnet-v2-1"
+        ## This handles other chains which tend to follow a pattern, with some chains sometimes missing depending on the subgraph
+        else:
+            ## balancer
+            if chain not in NO_BALANCER_SUBGRAPH:
+                balancer[chain] = f"https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-{chain}-v2"
+            else:
+                balancer[chain] = None
+            ## gauges
+            if chain not in NO_GAUGE_SUBGRAPH:
+                gauges[chain] = f"https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-gauges-{chain}"
+            else:
+                gauges[chain] = None
+            ## aura
+            if chain not in NO_AURA_SUBGRAPH:
+                aura[chain] = f"https://graph.aura.finance/subgraphs/name/aura/aura-{chain}-v2-1"
+            else:
+                aura[chain] = None
+        ## Overrides that are known to work
+        aura['zkevm'] = "https://subgraph.satsuma-prod.com/ab0804deff79/1xhub-ltd/aura-finance-zkevm/api"
+        gauges["zkevm"] = "https://api.studio.thegraph.com/query/24660/balancer-gauges-polygon-zk/version/latest"
+        balancer['zkevm'] = 'https://api.studio.thegraph.com/query/24660/balancer-polygon-zk-v2/version/latest'
+
+
+class GraphQueries:
+    def get_first_block_after_utc_timestamp(self, timestamp: int):
+        result = self.fetch_graphql_data(self.GET_FIRST_BLOCK_AFTER_TIMESTAMP, {"timestamp": int(timestamp)})
+        return int(result["data"]["blocks"][0]["number"])
+
+    def fetch_graphql_data(self, query_object, variables=None):
+        if variables:
+            response = requests.post(query_object["endpoint"],
+                                     json={'query': query_object["query"], 'variables': variables})
+        else:
+            response = requests.post(query_object["endpoint"], json={'query': query_object["query"]})
+        try:
+            response.raise_for_status()
+        except requests.HTTPError as e:
+            raise GraphQLRequestError(f"HTTP Error: {e}")
+        data = response.json()
+        if 'errors' in data:
+            raise GraphQLRequestError(f"Error: {data['errors']}")
+        return data
+
+    def __init__(self, chain):
+        self.chain = chain
+        
+        ## TODO add asert that result list is < 1000 due to first: 1000 in the query
+        self.BALANCER_POOL_SHARES_QUERY = {"endpoint": GraphEndpoints.balancer[chain], "query": """
```

Comment:
Can we refactor this a bit more? Ideally, all queries are stored in nicely separated constants files for each protocol and version and can by dynamically loaded including variables. Not sure if that is possible with your current setup but in the future this class file will be unmanageable. 

I would even argue to have a base class and extendables for balancer v2, gauges, aura etc.

Reply:
3f0a354